doc/source/whatsnew/v0.25.0.rst

```diff
@@ -158,6 +158,7 @@ I/O
 ^^^
 
 - Fixed bug in missing text when using :meth:`to_clipboard` if copying utf-16 characters in Python 3 on Windows (:issue:`25040`)
+- Bug in :func:`read_json` that tries to infer dtypes by default, even for ``orient='table'`` which already defines dtypes in its schema (:issue:`21345`)
```

Comment:
Hmm so I'm slightly confused about the intention here. The PR title and issue suggest we are only fixing something for the table orient, the whatsnew note here is somewhat ambiguous but then the implementation itself looks to be making an API change for all readers (though lacking tests outside of orient).

Shouldn't our scope here just be limited to the table orient? I understand the default value of `True` for reading makes that tricky but if we can't get around that this needs to be communicated as a wider API breaking change with testing for other orients I think

Reply:
@WillAyd to say it in a nutshell, the only thing this PR does is, for `orient='table'`, the argument `dtype` is no longer applicable, as dtypes are inferred from the 'schema' of the JSON document. This is done, internally, by setting `dtype=False` for `orient='table'`, but this is irrelevant and the user has no need to know about this internal implementation.

And I would like to point out that nothing has changed for the rest of the orients, as `dtype` can be passed and it defaults to True, identically as it was the case before this PR.
-

Reply:
I am going to rephrase the whatsnew note.